File: apps/web/lib/mocks/evaluation.ts
/** Preview-only evaluation payload until GET /sessions/{id}/evaluation exists. */

export interface EvaluationPreview {
  overall_score: number;
  dimension_scores: {
    communication: number;
    technical: number | null;
    relevance: number;
    structure: number;
    confidence: number;
    conciseness: number;
    problem_solving: number | null;
  };
  strengths: string[];
  weaknesses: string[];
  priority_improvements: Array<{
    area: string;
    priority: number;
    recommendation: string;
    rationale: string;
  }>;
  answer_evaluations: Array<{
    question: string;
    answer_excerpt: string;
    feedback: string;
    score: number;
  }>;
  better_answers: Array<{
    question: string;
    example: string;
  }>;
  practice_recommendations: Array<{
    title: string;
    instructions: string;
    success_criteria: string;
  }>;
}

export function buildPreviewEvaluation(sessionTitle: string): EvaluationPreview {
  return {
    overall_score: 76,
    dimension_scores: {
      communication: 78,
      technical: 74,
      relevance: 81,
      structure: 70,
      confidence: 77,
      conciseness: 65,
      problem_solving: 72,
    },
    strengths: [
      "You opened with a clear diagnostic sequence when explaining how you would debug latency.",
      "You referenced concrete observability signals (metrics and traces) rather than staying abstract.",
    ],
    weaknesses: [
      "The answer spends significant time on process before stating the expected user or business impact.",
      "Trade-offs and rollback criteria were mentioned briefly but not tied to a specific incident outcome.",
    ],
    priority_improvements: [
      {
        area: "structure",
        priority: 1,
        recommendation:
          "Lead with the outcome or hypothesis in one sentence, then walk through your debugging steps.",
        rationale:
          "Interviewers at senior level expect impact-first framing before implementation detail.",
      },
    ],
    answer_evaluations: [
      {
        question: "How would you debug elevated API latency?",
        answer_excerpt:
          "I would inspect metrics, traces, and recent deploys before narrowing to a bottleneck.",
        feedback:
          "Strong start on observability tooling. Add a one-sentence statement of what 'fixed' looks like (e.g., restored p95 under 200ms) before expanding.",
        score: 74,
      },
    ],
    better_answers: [
      {
        question: "How would you debug elevated API latency?",
        example:
          "I would first confirm the user-facing symptom—p95 checkout latency jumped from 180ms to 900ms—then compare APM traces and recent deploys to isolate the slow dependency before rolling forward a targeted fix.",
      },
    ],
    practice_recommendations: [
      {
        title: "Outcome-first retake drill",
        instructions: `Re-answer the primary question from "${sessionTitle}" with sentence one stating the business or user impact.`,
        success_criteria:
          "First sentence names a measurable outcome; remaining sentences support it with actions taken.",
      },
    ],
  };
}

File: apps/api/app/schemas/evaluation.py
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, Field

EvaluationStatusLiteral = Literal["pending", "running", "completed", "failed", "unavailable"]


class EvaluationDimensionScores(BaseModel):
    communication: float | None = None
    technical: float | None = None
    relevance: float | None = None
    structure: float | None = None
    confidence: float | None = None
    conciseness: float | None = None
    problem_solving: float | None = None


class EvaluationImprovementItem(BaseModel):
    area: str
    priority: int = Field(ge=1, le=5)
    recommendation: str
    rationale: str | None = None


class PracticeRecommendationItem(BaseModel):
    title: str
    instructions: str
    success_criteria: str


class BetterAnswerItem(BaseModel):
    question: str | None = None
    example: str


class AgentStatusItem(BaseModel):
    agent_name: str
    status: str
    skipped: bool = False
    summary: str | None = None


class SessionEvaluationData(BaseModel):
    overall_score: float
    dimension_scores: EvaluationDimensionScores
    strengths: list[str] = Field(default_factory=list)
    weaknesses: list[str] = Field(default_factory=list)
    priority_improvements: list[EvaluationImprovementItem] = Field(default_factory=list)
    better_answers: list[BetterAnswerItem] = Field(default_factory=list)
    practice_recommendations: list[PracticeRecommendationItem] = Field(default_factory=list)
    did_well: list[str] = Field(default_factory=list)
    should_improve: list[str] = Field(default_factory=list)
    judge_summary: str | None = None
    coach_summary: str | None = None
    hire_recommendation: str | None = None
    weights_applied: dict[str, Any] = Field(default_factory=dict)


class SessionEvaluationResponse(BaseModel):
    session_id: UUID
    session_status: str
    evaluation_status: EvaluationStatusLiteral
    evaluation: SessionEvaluationData | None = None
    error_message: str | None = None
    agents: list[AgentStatusItem] = Field(default_factory=list)
